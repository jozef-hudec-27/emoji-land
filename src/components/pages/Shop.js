import EmojiList from '../emoji/EmojiList'

export default function Shop() {
  return (
    <section id="shop">
      <EmojiList />
    </section>
  )
}
