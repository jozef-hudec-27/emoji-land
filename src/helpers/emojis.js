const randomPrice = () => {
  return Math.floor(Math.random() * 10) + 1
}

const emojis = [
  {
    emoji: 'ðŸ˜€',
    price: randomPrice(),
    name: 'Grinning Face',
    description:
      'A yellow face with simple, open eyes and a broad, open smile, showing upper teeth and tongue on some platforms. Often conveys general pleasure and good cheer or humor.',
  },
  {
    emoji: 'ðŸ˜‚',
    price: randomPrice(),
    name: 'Face with Tears of Joy',
    description:
      'A yellow face with a big grin, uplifted eyebrows, and smiling eyes, each shedding a tear from laughing so hard. Widely used to show something is funny or pleasing.',
  },
  {
    emoji: 'ðŸ™‚',
    price: randomPrice(),
    name: 'Slightly Smiling Face',
    description:
      'A yellow face with simple, open eyes and a thin, closed smile. Conveys a wide range of positive, happy, and friendly sentiments. Its tone can also be patronizing, passive-aggressive, or ironic, as if saying This is fine when itâ€™s really not.',
  },
  {
    emoji: 'ðŸ™ƒ',
    price: randomPrice(),
    name: 'Upside-Down Face',
    description:
      'Commonly used to convey irony, sarcasm, joking, or a sense of goofiness or silliness. Its intent can be similar to the bemused Oh well! of ðŸ¤· Person Shrugging or the shruggie emoticon, Â¯_(ãƒ„)_/Â¯.',
  },
  {
    emoji: 'ðŸ« ',
    price: randomPrice(),
    name: 'Melting Face',
    description:
      'Can be used literally to talk about extreme heat. Can also be used metaphorically to talk about embarrassment, shame, a slowly sinking sense of dread, or feeling overwhelmed in some way.',
  },
  {
    emoji: 'ðŸ˜‰',
    price: randomPrice(),
    name: 'Winking Face',
    description:
      'A yellow face with a slight smile or open mouth shown winking, usually its left eye. May signal a joke, flirtation, hidden meaning, or general positivity. Tone varies, including playful, affectionate, suggestive, or ironic.',
  },
  {
    emoji: 'ðŸ¥°',
    price: randomPrice(),
    name: 'Smiling Face with Hearts',
    description:
      'A yellow face with smiling eyes, a closed smile, rosy cheeks, and several hearts floating around its head. Expresses a range of happy, affectionate feelings, especially being in love. The placement of the hearts varies slightly by platform.',
  },
  {
    emoji: 'ðŸ˜¢',
    price: randomPrice(),
    name: 'Crying Face',
    description:
      'A yellow face with raised eyebrows and a slight frown, shedding a single, blue tear from one eye down its cheek. May convey a moderate degree of sadness or pain, usually less intensely than ðŸ˜­ Loudly Crying Face.',
  },
  {
    emoji: 'ðŸ˜ ',
    price: randomPrice(),
    name: 'Angry Face',
    description:
      'Conveys varying degrees of anger, from grumpiness and irritation to disgust and outrage. May also represent someone acting tough or being mean.',
  },
  {
    emoji: 'ðŸ˜ˆ',
    price: randomPrice(),
    name: 'Smiling Face with Horns',
    description:
      'A face, usually purple, with devil horns, a wide grin, and eyes and eyebrows scrunched downward in the same manner as ðŸ˜  Angry Face on most platforms. Googleâ€™s design is red and Facebookâ€™s has black horns and green eyes.',
  },
]

const findEmoji = (identificator) => {
  for (let i = 0; i < emojis.length; i++) {
    const { emoji, name, description } = emojis[i]
    if ([emoji, name, description].includes(identificator)) {
      return emojis[i]
    }
  }
}

export { emojis, findEmoji }
