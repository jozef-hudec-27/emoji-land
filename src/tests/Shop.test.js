import { render } from '@testing-library/react'
import Shop from '../components/pages/Shop'
import { emojis } from '../helpers/emojis'
import { BrowserRouter } from 'react-router-dom'

describe('Shop component', () => {
  it(`renders ${emojis.length} emoji cards`, () => {
    const { container } = render(
      <BrowserRouter>
        <Shop />
      </BrowserRouter>
    )

    expect(container).toMatchSnapshot()
  })
})
