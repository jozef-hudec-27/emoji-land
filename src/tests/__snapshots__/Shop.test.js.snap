// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shop component renders 10 emoji cards 1`] = `
<div>
  <section
    id="shop"
  >
    <div
      id="emoji-list"
    >
      <a
        class="emoji-card flexbox flex-center"
        href="/emoji/ðŸ˜€"
      >
        <h3>
          ðŸ˜€
        </h3>
      </a>
      <a
        class="emoji-card flexbox flex-center"
        href="/emoji/ðŸ˜‚"
      >
        <h3>
          ðŸ˜‚
        </h3>
      </a>
      <a
        class="emoji-card flexbox flex-center"
        href="/emoji/ðŸ™‚"
      >
        <h3>
          ðŸ™‚
        </h3>
      </a>
      <a
        class="emoji-card flexbox flex-center"
        href="/emoji/ðŸ™ƒ"
      >
        <h3>
          ðŸ™ƒ
        </h3>
      </a>
      <a
        class="emoji-card flexbox flex-center"
        href="/emoji/ðŸ« "
      >
        <h3>
          ðŸ« 
        </h3>
      </a>
      <a
        class="emoji-card flexbox flex-center"
        href="/emoji/ðŸ˜‰"
      >
        <h3>
          ðŸ˜‰
        </h3>
      </a>
      <a
        class="emoji-card flexbox flex-center"
        href="/emoji/ðŸ¥°"
      >
        <h3>
          ðŸ¥°
        </h3>
      </a>
      <a
        class="emoji-card flexbox flex-center"
        href="/emoji/ðŸ˜¢"
      >
        <h3>
          ðŸ˜¢
        </h3>
      </a>
      <a
        class="emoji-card flexbox flex-center"
        href="/emoji/ðŸ˜ "
      >
        <h3>
          ðŸ˜ 
        </h3>
      </a>
      <a
        class="emoji-card flexbox flex-center"
        href="/emoji/ðŸ˜ˆ"
      >
        <h3>
          ðŸ˜ˆ
        </h3>
      </a>
    </div>
  </section>
</div>
`;
